### PR TITLE
WIP: Manuscript refresh

### LINKED ISSUE #1162: Autosave feature broken
When uploading a supporting file the autosave does not work. See discussion in agenda 4/12/2018

https://docs.google.com/document/d/1IEYlmkzyBmxEFhn8U0kM8F5cl5OO-2jEpJmVV4VGZus/edit

### LINKED ISSUE #1180: Manuscript Refresh
Quick fix for #1162

Depends on #1087

### PR DESCRIPTION
**Pull request by Cooryd**
#### Background

Currently there is a bug in the supporting file uploads mutation that means the manuscript object may contain stale data if an auto-save occurs while a file is still uploading.

As a quick fix for this (pending a potential job queue service which would allow us to queue any changes) we should refresh the object before we make changes to its data but after we have uploaded the file to s3 to make sure any other simultaneous edits are not over-written.

This work adds a refresh function to the `Manuscript` class and implements it before the manuscript save in the `supportingUpload` and `manuscriptUpload` mutations.

Still leaves #1162 to complete

#### Any relevant tickets

closes #1180

#### How has this been tested?

Please indicate the need for this change to be tested. If not please justify, if so then at what levels. **Reviewers** ensure that these test requirements are also reviewed.

- [ ] Unit / Integration tests
- [ ] Browser tests
- [ ] End2end tests

#### UI Changes

- [ ] Has been reviewed against Designs
- [ ] Necessary updates to styleguide

### CODE REVIEW

**Comment by damnhipster** [comment]
Dependant on #1087

File: server/xpub-model/entities/file/index.js [hunk 1]
```diff
@@ -66,6 +66,16 @@ class File extends BaseModel {
...
+  static async findByManuscriptId(manuscriptId) {
+    const files = await this.query().where({
+      manuscript_id: manuscriptId,
+    })
+    if (!files) {
+      throw new Error(`file with manuscript_id ${manuscriptId} not found`)
+    }
+    return files
+  }
+
```
Comment (damnhipster) [inline server/xpub-model/entities/file/index.js]: This is duplicated, I think the merge failed. Will attempt to fix it.

File: server/xpub-server/entities/manuscript/resolvers/uploadSupportingFiles.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    throw err
+  }
+  manuscript.refresh()
+  manuscript.files.push(fileEntity)
+  await manuscript.save()
+
+  return manuscript
+}
+
+module.exports = uploadSupportingFiles
```
Comment (damnhipster) [inline server/xpub-server/entities/manuscript/resolvers/uploadSupportingFiles.js]: This file is also duplicated.

**Comment by damnhipster** [comment]
I think it's worth creating a new PR for these changes, the history on this PR seems out of place and it's making it difficult to review. Will get started on it.

**Comment by diversemix** [comment]
Been replaced with #1184

**Pull request closed without merging**